### ISSUE TITLE
Home View: message-timestamp changes to wrong value after closing & reopening signal4ios 

### ISSUE THREAD

**Issue by five-c-d** [signalapp/Signal-iOS#4143]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
<ins>Updated.</ins>  Somewhat-intermittently but quite often, message-timestamps <ins>in the conversation-list as opposed to in the chat-bubbles</ins> render with known-to-be-inaccurate values.  This behavior was observed on two different iPhones, both Xs running 12.1.4, and with chats when connected to multiple different contacts. The devices are reportedly configured to SetAutomatically for their clock-time and timezones.  There are no linked-devices running signal4desktop downstream.  

### Steps to reproduce
- set disappearing-messages to 5-minutes (for all chats) 
- <s>&nbsp;when a message arrives, manually delete it as soon as you have read it&nbsp; </s>
- <s>&nbsp;continue this usage-pattern for a long time&nbsp;</s>
- <s>&nbsp;approximate late December or early January, begin to notice inaccurate timestamps (not sure what signal4ios version shipped at that time... or whether the change is the result of Apple updates perhaps... or a combination) &nbsp;</s>
- <ins>watch the timestamps of your contacts as rendered in the conversation-list, aka the "lastmod" </ins>
- <ins>when you send or receive a message, the lastmod will be updated, accurately </ins>
- <ins>however, if you close signalapp and reopen it later, AFTER all the chat-messages have disappeared but before any new ones are sent or received with that contact, </ins>
- <ins>the lastmod(s) <a href="https://community.signalusers.org/t/time-stamp-issue/6472/8">will have timewarped to an earlier point in time</a> (profile-load-operation?)  </ins>

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
- in some cases the inaccuracy is "mild" ... a message sent from iPhone#1 to contact#xxx633 will in reality be sent ~8:19pm, but after closing and reopening signal4ios it will render with the sent-timestamp of 7:31pm instead.   Note that the debuglog SHOWS that a message WAS in fact sent at 7:31pm... but not **that** particular message.  
-  in some cases the inaccuracy is "severe" ... a message sent from iPhone#1 to contact#xxx370 at ~7am Thursday morning, after closing and reopening signal4ios, will render the sent-timestamp of 'WED' instead (i.e. the day earlier).  

**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)
- when a message is sent at HH:MM the chat-bubble metadata <ins>and the conversation-list "lastmod"</a> should always reflect that, with no spooky time-warps  
- <ins> the conversation-list lastmod should be monotonically increasing, and should show the last time you interacted with the contact... even if the most recent message happens to have already been disappeared during the disappearing-messages-job</ins> 

### Screenshots

The iphone-topbar-clock is clearly showing the correct localtime, and timestamp-metadata in the signal4ios screen (conversation-list entry rather than inside the chat-history methinks) is showing something known-to-be-wrong.  

*  https://community.signalusers.org/t/time-stamp-issue/6472/8 , two screenshots to match the two debuglogs (below)
*  more screenshots available if needed 

### Device info

**Device**: iPhone Xs 

**iOS version**: 12.1.4 

**Signal version:** 2.36.1.0

### Link to debug log

*  https://debuglogs.org/bb5afdf3d38c875362b3caaf1eb7668f85864f3fb4a1960ab413bbb3b6ca705e.zip , message sent ~~8:15pm evening localtime, when looking at it again 8:29pm localtime, inaccurately claims it was sent "7:31pm" 

* https://debuglogs.org/e000a9c13fea412207d508fcfc43596d4d14aedc1b2e615405138b56d194d6d7.zip , message sent to xxx370 at ~~7:30am thursday morning localtime, when looking at it again 7:42am localtime, inaccurately claims it was sent "WED" 

* third debuglog available if needed 

### Other info 

*  https://github.com/signalapp/Signal-iOS/issues/3447 , Problems with timestamp , in particular this screenshot https://user-images.githubusercontent.com/34192761/40183659-ab9bb062-59ee-11e8-8875-461f5cfe75a1.jpeg , closed as probably-a-clock-config-issue (but may not have been?).  
*  https://github.com/signalapp/Signal-iOS/issues/3511 ,  Message order on iOS and Windows not aligned , May.2018 , <s> &nbsp; might be related/relevant.  &nbsp;  </s>  <ins> Seems unrelated now that further investigation has revealed what is happening </ins> 
*  More details at this forum-thread, https://community.signalusers.org/t/time-stamp-issue/6472 

Note that the same iPhone#1 is also sometimes exhibiting a data-loss problem, as follows:  

*  new message arrives, notification-popup appears 
*  enduser does NOT mark-as-read the message within the popup 
*  however, very intermittently, five minutes later the disappearing-messages timeout kicks in, despite the enduser NOT yet having actually read the message 
*  they know it *was* there because of the notification, but not what was in it... frustrating
*  hard to capture a debuglog because this happens rarely, and because it is showing "something that did not happen"  

<s> &nbsp; My suspicion is that there is some kind of problem in the local database, where the timestamps are getting pulled from the wrong row (and/or put into the wrong row), possibly?  &nbsp; </s>  <ins> See explanation at the signalUsers thread, the problem is that the conversation-list does not 'remember' timestamps of recent messages, once those messages have disappeared, causing the lastmod to sometimes timewarp to an earlier value whenever all recently-exchanged messages have expired and/or been manually deleted.  

Since not many other iPhone endusers seem to be reporting/experiencing these kinds of timestamp-time-warp problems, might be specific to the iOS version and device-model, <s>or MIGHT be specific</s> <ins>and is definitely related</ins> to the rare usage-pattern (5-min disappear *plus* <ins>exacerbated by</ins> manual delete almost every time).  <ins> Should be possible to reproduce with simply a single contact setup with disappearing-messages on a short timer, however.</ins>

**Comment by five-c-d** [comment]
TLDR:  Alice powers up her new iphone at 8am, immediately installs signalapp, registers, and sends & receives a couple test-messages with Bob.  Alice sets the disappearing-messages timer to 5-minutes in their chat at 9am.   At 10am Bob sends Alice a message.  At 10:01 when Alice looks at her conversation-list, she sees "Bob ... themsg ... 10:00am"  

However, when Alice looks again at the conversation list an hour later at 11am, that 10am message from Bob has already disappeared... and Alice now sees "Bob ... somethingOlder ... backwardsTimewarpHH:MM" even though the most recent interaction with Bob was 10:00am.

**Comment by michaelkirk-signal** [comment]
As @greyson-signal mentioned in https://github.com/signalapp/Signal-Android/issues/8662, this is the intended behavior.
